src/meta/server_state.cpp

```diff
@@ -1259,6 +1259,105 @@ void server_state::drop_app(dsn::message_ex *msg)
     }
 }
 
+void server_state::do_app_rename(configuration_rename_app_rpc rpc)
+{
+    zauto_write_lock l(_lock);
+
+    const auto &old_app_name = rpc.request().old_app_name;
+    const auto &new_app_name = rpc.request().new_app_name;
+
+    auto target_app = get_app(old_app_name);
+    auto app_id = target_app->app_id;
+
+    auto ainfo = *(reinterpret_cast<app_info *>(target_app.get()));
+    ainfo.app_name = new_app_name;
+    auto app_path = get_app_path(*target_app);
+
+    target_app->app_name = new_app_name;
+    _exist_apps.emplace(new_app_name, target_app);
```

Comment:
Why not add new app name after updating remote storage ? That will be an atomic operation including both adding new table name and removing old table name.

Reply:
Maybe a new table named new_app_name has been created successfully，So add it firstly.